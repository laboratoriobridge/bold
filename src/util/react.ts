import { MutableRefObject, Ref } from 'react'

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}
