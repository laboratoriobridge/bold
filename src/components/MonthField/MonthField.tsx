import { Options as PopperOptions } from '@popperjs/core'
import React, { useEffect, useState } from 'react'
import { usePopper } from 'react-popper'
import { useStyles } from '../../styles/hooks/useStyles'
import { Theme } from '../../styles/theme/createTheme'
import { composeHandlers, composeRefs } from '../../util/react'
import { FocusManagerContainer } from '../FocusManagerContainer'
import { MonthPicker, MonthPickerProps, ReferenceMonth } from '../MonthPicker'
import { disabledByMonth, isSameReferenceMonth } from '../MonthRangePicker/util'
import { isValidReferenceMonth } from '../Calendar/util'
import { MonthInput, MonthInputProps } from './MonthInput'

export interface MonthFieldProps extends MonthInputProps {
  minMonth?: ReferenceMonth
  maxMonth?: ReferenceMonth
  popperProps?: PopperOptions
  monthPickerProps?: Partial<MonthPickerProps>
}

export function MonthField(props: MonthFieldProps) {
  const { value, popperProps, inputRef, onChange, onIconClick, maxMonth, minMonth, ...rest } = props

  const { classes, css } = useStyles(createStyles)

  const [open, setOpen] = useState(false)

  const [anchorRef, setAnchorRef] = useState<HTMLInputElement>()
  const [popperRef, setPopperRef] = useState<HTMLDivElement>()

  const [visibleMonth, setVisibleMonth] = useState<ReferenceMonth>({
    month: new Date().getMonth(),
    year: new Date().getFullYear(),
  })

  useEffect(() => {
    setVisibleMonth(
      isValidReferenceMonth(value)
        ? value
        : {
            month: new Date().getMonth(),
            year: new Date().getFullYear(),
          }
    )
  }, [value])

  const {
    styles: { popper: popperStyle },
    attributes: { placement },
  } = usePopper(anchorRef, popperRef, { ...popperProps, placement: 'bottom-start' })

  const handleVisibleMonthChange = (month: ReferenceMonth) => setVisibleMonth(month)

  const handleInputChange = (month: ReferenceMonth) => {
    onChange(month)
  }

  const handleMonthClick = (month: ReferenceMonth) => {
    setOpen(false)
    onChange(month)
  }

  const handleInputIconClick = () => setOpen(true)
  const handleFocusIn = () => setOpen(true)
  const handleFocusOut = () => setOpen(false)

  return (
    <FocusManagerContainer onFocusIn={handleFocusIn} onFocusOut={handleFocusOut}>
      <MonthInput
        inputRef={composeRefs(setAnchorRef, inputRef)}
        value={value}
        icon='calendarOutline'
        onIconClick={composeHandlers(handleInputIconClick, onIconClick)}
        onChange={handleInputChange}
        {...rest}
      />

      {open && (
        <div
          ref={setPopperRef}
          className={css(classes.popup, popperStyle as any)}
          data-placement={placement}
          data-testid='MonthField.popup'
        >
          <MonthPicker
            visibleMonth={visibleMonth}
            onVisibleMonthChange={handleVisibleMonthChange}
            onMonthClick={handleMonthClick}
            isDisabled={disabledByMonth(minMonth, maxMonth)}
            modifiers={{
              selected: (month: ReferenceMonth) => value && isSameReferenceMonth(value, month),
            }}
            {...props.monthPickerProps}
          />
        </div>
      )}
    </FocusManagerContainer>
  )
}

const createStyles = (theme: Theme) => ({
  popup: {
    zIndex: theme.zIndex.popper,
  },
})
