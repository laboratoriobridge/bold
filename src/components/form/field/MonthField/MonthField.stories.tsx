import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react'
import React from 'react'

import { withForm } from '../../../../stories-addons'

import { MonthField } from './MonthField'

storiesOf('Form|MonthField', module)
  .addDecorator(withForm())
  .add('default', () => <MonthField onChange={action('changed')} name='month' />)
