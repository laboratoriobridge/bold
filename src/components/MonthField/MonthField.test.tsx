import { fireEvent, render } from '@testing-library/react'
import React from 'react'

import { format, isValidInput, MonthField } from './MonthField'

const fn = jest.fn()

describe('Component presentation', () => {
  it('should render correctly', () => {
    const { container } = render(<MonthField value={{ year: 2019, month: 1 }} onChange={fn} />)
    expect(container).toMatchSnapshot()
  })
  it('should render MonthPicker on focus', () => {
    const { container, getByTitle } = render(<MonthField onChange={fn} title='Month Picker Input' />)

    const beforeFocus = container.querySelector('[data-visible="true"]')
    expect(beforeFocus).toBeNull()

    fireEvent.focus(getByTitle('Month Picker Input'))

    const afterFocus = container.querySelector('[data-visible="true"]')
    expect(afterFocus).not.toBeNull()
  })
})

describe('Component behavior', () => {
  it('should call "onChange" when a valid date is typed', () => {
    const { getByTitle } = render(<MonthField onChange={fn} title='Month Picker Input' />)

    fireEvent.change(getByTitle('Month Picker Input'), { target: { value: '08/2016' } })
    expect(fn).toHaveBeenCalledWith({ month: 7, year: 2016 })
  })
  it('should hide MonthPicker when a month is picked', () => {
    const { container, getByTitle, getByText } = render(<MonthField onChange={fn} title='Month Picker Input' />)

    fireEvent.focus(getByTitle('Month Picker Input'))
    fireEvent.click(getByText('Jan'))

    const afterSelect = container.querySelector('[data-visible="false"]')
    expect(afterSelect).not.toBeNull()
  })
})

describe('Other functions', () => {
  it('format should return null when argument is null', () => {
    let formatedValue = format(null)
    expect(formatedValue).toBeNull()

    formatedValue = format({ month: null, year: null })
    expect(formatedValue).toBeNull()
  })
  it('format should return null when months argument is null', () => {
    const formatedValue = format({ month: null, year: 2000 })
    expect(formatedValue).toBeNull()
  })
  it('format should return null when years argument is null', () => {
    const formatedValue = format({ month: 1, year: null })
    expect(formatedValue).toBeNull()
  })
  it('format should include 0 when month = 1', () => {
    const formatedValue = format({ month: 1, year: 2000 })
    expect(formatedValue).toBe('02/2000')
  })
  it('format should include 0 when month = 8', () => {
    const formatedValue = format({ month: 8, year: 2000 })
    expect(formatedValue).toBe('09/2000')
  })
  it('format should not include 0 when month > 8', () => {
    const formatedValue = format({ month: 9, year: 2000 })
    expect(formatedValue).toBe('10/2000')
  })
  it('isValidInput should accept mm/aaaa format', () => {
    expect(isValidInput('00/0000')).toBe(true)
  })
  it('isValidInput should not accept m/aaaa format', () => {
    expect(isValidInput('0/0000')).toBe(false)
  })
  it('isValidInput should not accept mm/aa format', () => {
    expect(isValidInput('00/00')).toBe(false)
  })
  it('isValidInput should not accept null', () => {
    expect(isValidInput(null)).toBe(false)
  })
})
