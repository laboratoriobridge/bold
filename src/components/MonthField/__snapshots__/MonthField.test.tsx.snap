// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component presentation should render correctly 1`] = `
.emotion-8 {
  display: block;
}

.emotion-6 {
  position: relative;
}

.emotion-6 input {
  padding-right: 4.5rem;
}

.emotion-2 {
  position: absolute;
  right: 2.5rem;
  background: transparent;
  border: none;
  cursor: pointer;
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: 0;
  color: #696979;
  padding: 1px 6px;
}

.emotion-2:hover {
  color: #D01E29;
}

.emotion-5 {
  position: absolute;
  background-color: #F0F0F5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.25rem;
  width: 2.5rem;
  top: 1px;
  bottom: 1px;
  right: 1px;
}

.emotion-0 {
  background-color: #FFFFFF;
  border: solid 1px #8F8FA2;
  border-radius: 2px;
  color: #3A3A47;
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1rem;
  padding: calc(0.5rem - 1px) 0.5rem;
  width: 100%;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-duration: .2s;
  transition-duration: .2s;
}

.emotion-0[type="search"]::-webkit-search-decoration {
  display: none;
}

.emotion-0[type="search"]::-webkit-search-cancel-button {
  display: none;
}

.emotion-0[type="search"]::-webkit-search-results-button {
  display: none;
}

.emotion-0[type="search"]::-webkit-search-results-decoration {
  display: none;
}

.emotion-0:required {
  box-shadow: none;
}

.emotion-0::-webkit-input-placeholder {
  color: #AAAAB9;
}

.emotion-0::-moz-placeholder {
  color: #AAAAB9;
}

.emotion-0:-ms-input-placeholder {
  color: #AAAAB9;
}

.emotion-0::placeholder {
  color: #AAAAB9;
}

.emotion-0:disabled {
  border-color: #D3D4DD;
  background-color: #F0F0F5;
}

.emotion-0:not(:disabled):hover {
  border-color: #7A7B8E;
}

.emotion-0:not(:disabled):focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-0:not(:disabled):active {
  border-color: #0069D0;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.09);
}

.emotion-1 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-4 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #3A3A47;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #3A3A47;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  border-radius: inherit;
}

.emotion-4 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-4:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-4:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-4:not(:disabled):hover {
  background-color: rgba(58,58,71,0.1);
}

.emotion-4:focus {
  box-shadow: none;
}

.emotion-43 {
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  z-index: 1200;
  padding: 0rem;
  visibility: hidden;
  opacity: 0;
}

.emotion-42 {
  background-color: #FFFFFF;
  display: inline-block;
  padding: 1rem;
  border: 1px solid #D3D4DD;
  box-shadow: 0 1px 5px 0 rgba(0,0,0,0.2),0 3px 1px -2px rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.14);
  border-radius: 2px;
}

.emotion-41 {
  margin: -0.25rem -0.25rem;
  display: grid;
  grid-template-columns: repeat(3,1fr);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-11 {
  text-align: center;
  margin: 0.25rem 0.25rem;
}

.emotion-10 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #3A3A47;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #3A3A47;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
}

.emotion-10 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-10:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-10:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-10:not(:disabled):hover {
  background-color: rgba(58,58,71,0.1);
}

.emotion-12 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5;
  color: #3A3A47;
  font-size: 0.875rem;
  font-weight: bold;
}

.emotion-17 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #3A3A47;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #3A3A47;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.75rem - 1px) 2.5rem;
  padding: calc(0.25rem - 1px) 1rem;
  -webkit-transition-property: background;
  transition-property: background;
  min-width: 70px;
}

.emotion-17 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-17:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-17:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-17:not(:disabled):hover {
  background-color: rgba(58,58,71,0.1);
}

.emotion-19 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #3A3A47;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #3A3A47;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.75rem - 1px) 2.5rem;
  padding: calc(0.25rem - 1px) 1rem;
  -webkit-transition-property: background;
  transition-property: background;
  min-width: 70px;
  background: #0069D0 !important;
  color: #FFFFFF;
}

.emotion-19 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-19:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-19:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-19:not(:disabled):hover {
  background-color: rgba(58,58,71,0.1);
}

<div>
  <div
    class="emotion-8"
  >
    <div
      class="emotion-8"
    >
      <div
        class="emotion-7"
      >
        <div
          class="emotion-6"
        >
          <input
            class="emotion-0"
            placeholder="mm/yyyy"
            type="text"
            value="02/2019"
          />
          <span
            class="emotion-2"
            role="button"
            tabindex="-1"
            title="Clear"
          >
            <svg
              aria-hidden="true"
              class="emotion-1"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M8.071 6.657l4.243 4.242 4.242-4.242c.464-.48.935-.48 1.415 0s.48.95 0 1.414l-4.243 4.243 4.243 4.242c.471.472.471.943 0 1.415-.472.471-.943.471-1.415 0l-4.242-4.243L8.07 17.97c-.46.482-.932.482-1.414 0-.482-.483-.482-.954 0-1.415l4.242-4.242L6.657 8.07c-.472-.47-.472-.942 0-1.414s.943-.472 1.414 0z"
              />
            </svg>
          </span>
          <span
            class="emotion-5"
          >
            <button
              class="emotion-4"
              tabindex="-1"
              type="button"
            >
              <span>
                <svg
                  aria-hidden="true"
                  class="emotion-1"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <defs>
                    <path
                      d="M12 0c.619 0 .95.27.995.812l.005.13V2h1c1.273 0 1.938.607 1.996 1.822L16 4v14c0 1.273-.607 1.938-1.822 1.996L14 20H2C.727 20 .062 19.393.004 18.178L0 18V4.004c0-1.275.607-1.942 1.822-2L2 2h1V1c-.002-.667.33-1 1-1 .621 0 .954.287.998.862L5.004 1v1H11V.941C11 .314 11.333 0 12 0zm2 8H2v10h12V8zm-8.998 6H6c.667 0 1 .333 1 1 0 .619-.287.95-.862.995L6 16H5c-.667.001-1-.332-1-1 0-.62.289-.952.864-.995L5.002 14H6zM11 14c.667 0 1 .333 1 1s-.333 1-1 1h-1c-.667.001-1-.332-1-1 0-.668.335-1.001 1.002-1zm-5-4c.667 0 1 .333 1 1 0 .619-.287.95-.862.995L6 12H5c-.667.001-1-.332-1-1 0-.62.289-.952.864-.995L5.002 10H6zm5 0c.667 0 1 .333 1 1 0 .619-.287.95-.862.995L11 12h-1c-.667.001-1-.332-1-1 0-.62.289-.952.864-.995l.138-.005H11zm3-6H2v2h12V4z"
                      id="calendar-outline_svg__a"
                    />
                  </defs>
                  <use
                    transform="translate(4 2)"
                    xlink:href="#calendar-outline_svg__a"
                  />
                </svg>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="emotion-43"
      data-visible="false"
      style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
    >
      <div
        class="emotion-42"
      >
        <div
          class="emotion-41"
        >
          <div
            class="emotion-11"
          >
            <button
              class="emotion-10"
              title="Previous year"
              type="button"
            >
              <span>
                <svg
                  aria-hidden="true"
                  class="emotion-1"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M14.707 16.294a1 1 0 11-1.414 1.414l-5-5a1 1 0 010-1.414l5.001-5.001a1 1 0 011.414 1.414l-4.294 4.294 4.293 4.293z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <span
              class="emotion-12"
            >
              2019
            </span>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-10"
              title="Next year"
              type="button"
            >
              <span>
                <svg
                  aria-hidden="true"
                  class="emotion-1"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M8.294 7.707a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5.001a1 1 0 01-1.415-1.414L12.587 12 8.294 7.707z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="January"
              type="button"
            >
              <span>
                Jan
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-19"
              title="February"
              type="button"
            >
              <span>
                Feb
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="March"
              type="button"
            >
              <span>
                Mar
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="April"
              type="button"
            >
              <span>
                Apr
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="May"
              type="button"
            >
              <span>
                May
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="June"
              type="button"
            >
              <span>
                Jun
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="July"
              type="button"
            >
              <span>
                Jul
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="August"
              type="button"
            >
              <span>
                Aug
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="September"
              type="button"
            >
              <span>
                Sep
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="October"
              type="button"
            >
              <span>
                Oct
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="November"
              type="button"
            >
              <span>
                Nov
              </span>
            </button>
          </div>
          <div
            class="emotion-11"
          >
            <button
              class="emotion-17"
              title="December"
              type="button"
            >
              <span>
                Dec
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
